src/analyzer/analyzersilence.cpp

```diff
@@ -0,0 +1,104 @@
+#include "analyzer/analyzersilence.h"
+
+const int kChannelCount = 2;
+const float kSilenceThreshold = -60.0f;
+
+AnalyzerSilence::AnalyzerSilence(UserSettingsPointer pConfig)
+    : m_pConfig(pConfig),
+      m_fThreshold(db2ratio(kSilenceThreshold)),
+      m_iSampleRate(0),
+      m_iTotalSamples(0),
+      m_iFramesProcessed(0),
+      m_bPrevSilence(true),
+      m_iSignalBegin(-1),
+      m_iSignalEnd(-1) {
+}
+
+AnalyzerSilence::~AnalyzerSilence() {
+}
+
+bool AnalyzerSilence::initialize(TrackPointer tio, int sampleRate, int totalSamples) {
+    Q_UNUSED(tio);
+
+    m_iSampleRate = sampleRate;
+    m_iTotalSamples = totalSamples;
+    m_iFramesProcessed = 0;
+    m_bPrevSilence = true;
+    m_iSignalBegin = 0;
+    m_iSignalEnd = 0;
+
+    return true;
+}
+
+bool AnalyzerSilence::isDisabledOrLoadStoredSuccess(TrackPointer tio) const {
+    Q_UNUSED(tio);
+
+    return false;
+}
+
+void AnalyzerSilence::process(const CSAMPLE *pIn, const int iLen) {
+    for (int i = 0; i < iLen; i += kChannelCount) {
```

Comment:
Right, a simple low pass will not work.
For replaygain, it is a moving window of 50 ms with a RMS calculation is used. This sounds reasonable to suppress detecting unbearable spikes. 

Using this along with the human ear compensation filter used by RG would be a high end solution. 

However I am in doubt If this has any value compared to the extra work and extra time for an analysis run.

If we find a smart way to combine RG detection with silence detection into a single analyser, that would be nice an probably even faster than the current solution.

We may also look to the beatdetection code what is considered as THE first beat, that would help to distinguish unusable noise from dance beats. 

All these considerations are pretty advanced,   so we may want to go with the current solution to collect experiences.